roles/core_configure/tasks/storage_fileset.yml

```diff
@@ -0,0 +1,147 @@
+---
+
+##################### Fileset configuration For CES Nodes #####################
+
+# First verifing if fileset already exists if not then creating Independent fileset linking it and printing it.
+
+- name: storage | Find existing filesystem(s)
+  shell: /usr/lpp/mmfs/bin/mmlsfs all -Y | grep -v HEADER | cut -d ':' -f 7 | uniq
+  register: scale_storage_existing_fs
+  changed_when: false
+  failed_when: false
+
+- name: Debug the Filesystem name
+  debug:
+    var: scale_storage_existing_fs.stdout
+  ignore_errors: yes
+  when: is_admin_node | default(false) == true
+
+- name: Check if fileset in file system already exists
+  shell: "/usr/lpp/mmfs/bin/mmlsfileset {{ scale_storage_existing_fs.stdout }} | tail -n +3 | awk '{print $1}' | grep -w {{ item.key }}"
+  loop: "{{ scale_storage[0].filesets | dict2items }}"
+  register: filesets_result
+  ignore_errors: yes
+  failed_when: filesets_result.rc == 2
+  when: is_admin_node | default(false) == true
+
+- name: Print return code for each fileset
+  debug:
+    msg: "Fileset {{ item.item.key }} - Return Code: {{ item.rc }}"
+  loop: "{{ filesets_result.results }}"
+  loop_control:
+    loop_var: item
+  when: is_admin_node | default(false) == true
+
+- name: Create new fileset
+  shell: "/usr/lpp/mmfs/bin/mmcrfileset {{ scale_storage_existing_fs.stdout }} {{ item.item.key }} --inode-space new"
+  loop: "{{ filesets_result.results }}"
+  register: created_fileset
+  when: is_admin_node | default(false) == true and item.rc != 0
+  run_once: true
+
+- name: Print return code for created filesets
+  debug:
+    msg: "Fileset_created {{ item.item.item.key }} - Return Code: {{ item.item.rc }}"
+  loop: "{{ created_fileset.results }}"
+  loop_control:
+    loop_var: item
+  when: is_admin_node | default(false) == true
+
+- name: Link new fileset
+  shell: "/usr/lpp/mmfs/bin/mmlinkfileset {{ scale_storage_existing_fs.stdout }} {{ item.item.item.key }} -J {{ scale_storage[0].defaultMountPoint }}/{{ item.item.item.key }}"
+  loop: "{{ created_fileset.results }}"
+  when: is_admin_node | default(false) == true and item.changed == true
+  run_once: true
+
+- name: Check filesets in file system
+  command: /usr/lpp/mmfs/bin/mmlsfileset {{ scale_storage_existing_fs.stdout }}
+  register: filesets
+  ignore_errors: yes
+  when: is_admin_node | default(false) == true
+
+- name: Debug the filesets in file system
+  debug:
+    var: filesets.stdout_lines
+  ignore_errors: yes
+  when: is_admin_node | default(false) == true
+
+# Setting up quota for filesets
```

Comment:
But i think Quota may not be applicable for all different set of deployment ,so it should be handled within some different boolean flag if user wants to set quota they can say , example: `scale_quota_enabled = true`

Reply:
Resolved